packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,150 @@
+import { Template, Match } from '../../../assertions';
+import * as logs from '../../../aws-logs';
+import * as s3 from '../../../aws-s3';
+import * as s3deploy from '../../../aws-s3-deployment';
+import * as cdk from '../../../core';
+import { CustomResourceConfig } from '../../lib/custom-resource-config/custom-resource-config';
+
+describe('when logging is undefined', () => {
+  test('when CustomResourceConfig.addLogRetentionLifetime is called, singleton-backed custom resource has new logGroup with retention period specified', () => {
```

Comment:
This is a bit wordy; it's difficult to understand at a glance. 

```suggestion
  test('addLogRetentionLifetime creates a new log group with the correct retention period if one does not already exist', () => {
```